internal/common/rules/RuleMatrix.go

```diff
@@ -0,0 +1,48 @@
+package rules
+
+import "gonum.org/v1/gonum/mat"
+
+type RuleMatrix struct {
+	ruleName          string
+	RequiredVariables []string
+	ApplicableMatrix  mat.Dense
+	AuxiliaryVector   mat.VecDense
+}
+
+var AvailableRules = map[string]RuleMatrix{}
+
+func registerNewRule(ruleName string, requiredVariables []string, applicableMatrix mat.Dense, auxiliaryVector mat.VecDense) *RuleMatrix {
+	if _, ok := AvailableRules[ruleName]; ok {
+		//Some form of anger at this point
```

Comment:
you can use `panic(...)`

Reply:
Sure, does panic like end the program?, cos I was thinking of a less severe penalty for trying to re-register a rule.

Reply:
...or return an error, but since this is something we want to do instantly at startup in infra, panic is encouraged

Reply:
yep, panic ends it, or you can return an `error` and let the caller handle it.